cli/src/pcluster/templates/cdk_builder_utils.py

```diff
@@ -625,6 +625,42 @@ def _build_policy(self) -> List[iam.PolicyStatement]:
                     ),
                 ]
             )
+            capacity_reservation_ids = self._config.capacity_reservation_ids
+            if self._config.capacity_reservation_ids:
+                policy.append(
+                    iam.PolicyStatement(
+                        actions=["ec2:RunInstances"],
+                        effect=iam.Effect.ALLOW,
+                        resources=[
+                            self._format_arn(
+                                service="ec2",
+                                resource=f"capacity-reservation/{capacity_reservation_id}",
+                            )
+                            for capacity_reservation_id in capacity_reservation_ids
+                        ],
+                    )
+                )
+            capacity_reservation_resource_group_arns = self._config.capacity_reservation_resource_group_arns
+            if capacity_reservation_resource_group_arns:
+                policy.extend(
+                    [
+                        iam.PolicyStatement(
+                            actions=["ec2:RunInstances"],
+                            effect=iam.Effect.ALLOW,
+                            resources=capacity_reservation_resource_group_arns,
+                        ),
+                        iam.PolicyStatement(
+                            actions=["ec2:CreateFleet"],
+                            effect=iam.Effect.ALLOW,
+                            resources=capacity_reservation_resource_group_arns,
+                        ),
+                        iam.PolicyStatement(
+                            actions=["resource-groups:ListGroupResources"],
+                            effect=iam.Effect.ALLOW,
+                            resources=capacity_reservation_resource_group_arns,
```

Comment:
why there is no for loop for `capacity_reservation_resource_group_arns` the same way there is for `capacity_reservation_ids`?

Reply:
Because `capacity_reservation_resource_group_arns` is a list of ARNs. 
But `capacity_reservation_ids` is a list of IDs. I have to change the ids to ARNs using for loop